### PR TITLE
Major Website Backend Cleanup

### PR DESCRIPTION
**Pull request by 719Ben**
Cleaned up all the website backend. Should be easier to send any type of message we need and should be less error prone.

### CODE REVIEW

File: website/conn_manager.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+	//Unregister all old clients
+	unregister chan *connection
+
+	//message that gets sent out to all the users
+	broadcast chan pushMessage
+}
+
+//abrastion out the chan to a connection, this makes it easier to add parts
+//laster and gets rid of (chan chan objects)
```
Comment (mguida22) [inline website/conn_manager.go]: this comment sounds kinda weird, also spelling on later

File: website/conn_manager.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+	//register the connection so that we send messages to it
+	hub.register <- newConnection
+
+	//Listen to the closing of the http connection via the CloseNotifier
+	notify := w.(http.CloseNotifier).CloseNotify()
+
+	//defer the function for when the connection closes
+	go func() {
+		<-notify
```
Comment (mguida22) [inline website/conn_manager.go]: what does this do? specifically the arrow going to nothing

File: website/conn_manager.go [hunk 3]
```diff
@@ -67,7 +67,11 @@ func (hub *ConHub) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	//defer the function for when the connection closes
 	go func() {
+
+		//this waits for something to hit the notify channel, it waits here
+		//until the connection closes and writes to the channel
 		<-notify
```
Comment (719Ben) [inline website/conn_manager.go]: @mguida22 a better explanation for you

**Comment by mguida22** [comment]
LGTM 🎱

File: website/conn_manager.go [hunk 4]
```diff
@@ -15,23 +15,23 @@ type ConHub struct {
...
 	register chan *connection
 
-	//Unregister all old clients
+	//a channel that is used to unregister old clients
 	unregister chan *connection
 
-	//message that gets sent out to all the users
+	//a channel that is used to message that gets sent out to all the users
 	broadcast chan pushMessage
 }
```
Comment (719Ben) [inline website/conn_manager.go]: @mguida22 better comments for you

**Pull request merged**